refactor(minifier): remove duplicated `typeof` comparison to non-strict equality compression

sapphi-red opened the pull request:
This is handled by `try_minimize_binary`.
https://github.com/oxc-project/oxc/blob/c2ca7a70a4b07f253de63b8220258d90e9fe0f38/crates/oxc_minifier/src/peephole/minimize_conditions.rs#L109

sapphi-red commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9010?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9011** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9011?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9010** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9010?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9010?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9009** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9009?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9008** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9008?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9010).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 10, 8:55 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
refactor(minifier): remove duplicated `typeof` comparison to non-strict equality compression (#9010)

This is handled by `try_minimize_binary`.
https://github.com/oxc-project/oxc/blob/c2ca7a70a4b07f253de63b8220258d90e9fe0f38/crates/oxc_minifier/src/peephole/minimize_conditions.rs#L109
